modyn/storage/src/internal/database/storage_database_connection.cpp

```diff
@@ -130,8 +139,6 @@ bool StorageDatabaseConnection::add_dataset(const std::string& name, const std::
   if (!add_sample_dataset_partition(name)) {
     FAIL("Partition creation failed.");
   }
-  session.close();
```

Comment:
in the `bool StorageDatabaseConnection::add_dataset` function there were some early `return` statements. If those `close` calls were mandatory we would have produced memory leaks already. I also looked up how `soci::session` works in terms of RAII. `close` looks very similar to the destructor, so I assume it will release the connections automatically when leaving the function scope.

[session::close](https://github.com/SOCI/soci/blob/884808c294e4809064650f3e5aed9102d8a68d56/src/core/session.cpp#L265)
[session:~session](https://github.com/SOCI/soci/blob/884808c294e4809064650f3e5aed9102d8a68d56/src/core/session.cpp#L210)

Reply:
Nice, then let's try to remove it. In case anything fails because of it, we can surely restore it.

Reply:
Okay, [here](https://github.com/eth-easl/modyn/actions/runs/8892578698/job/24423526911) it seems as if the changes cause some weird timeouts. I'll revert the changes.